### PR TITLE
Adding chime5 baseline recipe

### PR DESCRIPTION
**Pull request by jtrmal**
This is a joint work with @sw005320 and me
Dan, we are still working on the chain/nnet trainining but we thought we will go ahead and create PR now, so that at least the data preparation and gmm can go through review

### CODE REVIEW

File: egs/chime5/s5/run.sh [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+. ./cmd.sh
+. ./path.sh
+
+
+set -e # exit on error
+
+# chime5 main directory path
+# please change the path accordingly
+chime5_corpus=/export/corpora4/CHiME5
+json_dir=${chime5_corpus}/data/transcriptions
```
Comment (ShigekiKarita) [inline egs/chime5/s5/run.sh]: maybe  `json_dir=${chime5_corpus}/transcriptions`
there is no `data` dir in unzipped `CHiME5` dir

File: egs/chime5/s5/run.sh [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+audio_dir=${chime5_corpus}/data/audio
```
Comment (ShigekiKarita) [inline egs/chime5/s5/run.sh]: ditto

**Comment by jtrmal** [comment]
@danpovey @ShigekiKarita @sw005320 I'm addressing the comments, will push today -- I want to run the data preparation pipeline to make sure it works with the new corpus location

**Comment by sw005320** [comment]
@jtrmal @ShigekiKarita, I just fixed wrong path issues. Now I confirmed that it's working by state 16.

File: egs/chime5/s5/run.sh (hunk 1 above)
Comment (sw005320) [reply egs/chime5/s5/run.sh]: fixed

File: egs/chime5/s5/run.sh (hunk 2 above)
Comment (sw005320) [reply egs/chime5/s5/run.sh]: fixed

File: egs/chime5/s5/local/train_lms_srilm.sh [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+  -kndiscount1 -gt1min 0 -kndiscount2 -gt2min 2 -kndiscount3 -gt3min 2 -kndiscount4 -gt4min 3 -order 4 \
+  -text $tgtdir/train.txt -vocab $tgtdir/vocab -unk -sort -map-unk "$oov_symbol"
+
+if [ ! -z ${LIBLBFGS} ]; then
+  #please note that if the switch -map-unk "$oov_symbol" is used with -maxent-convert-to-arpa, ngram-count will segfault
+  #instead of that, we simply output the model in the maxent format and convert it using the "ngram"
+  echo "-------------------"
+  echo "Maxent 3grams"
+  echo "-------------------"
+  sed 's/'${oov_symbol}'/<unk>/g' $tgtdir/train.txt | \
```
[the remark this answers is not in the document]
Comment (jtrmal) [reply egs/chime5/s5/local/train_lms_srilm.sh]: That script is fairly general because there was a lot of experimentation with the training set size and yet, it's not general enough, as it contains some chime5 specific filtering (as it is probable, that the training set could contain duplicate transcriptions of the same utterance (via a different channel). IMO the MaxEnt models are usually the most robust way for arpa-format LM -- you don't have to mess with discounting and cut-offs, but they are not very widely accepted, so I wanted to provide a comparison against KN and GT baselines.

**Comment by jtrmal** [comment]
The non-addressed comments are related on TDNN/chain script, shinji is still waiting for his training to finish.

**Comment by sw005320** [comment]
@kamo-naoyuki which part are you talking about?

File: egs/chime5/s5/local/chain/tuning/run_tdnn_1e.sh [hunk 4]
```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+# 1e is as 1d but instead of the --proportional-shrink option, using
```
[the remark this answers is not in the document]
Comment (sw005320) [reply egs/chime5/s5/local/chain/tuning/run_tdnn_1e.sh]: Done

File: egs/chime5/s5/local/chain/tuning/run_tdnn_1e.sh [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+
+
+if [ $stage -le 13 ]; then
+  mkdir -p $dir
+  echo "$0: creating neural net configs using the xconfig parser";
+
+  num_targets=$(tree-info $tree_dir/tree |grep num-pdfs|awk '{print $2}')
+  learning_rate_factor=$(echo "print 0.5/$xent_regularize" | python)
+  opts="l2-regularize=0.05"
+  output_opts="l2-regularize=0.01"
```
[the remark this answers is not in the document]
Comment (sw005320) [reply egs/chime5/s5/local/chain/tuning/run_tdnn_1e.sh]: I just put "bottleneck-dim=320" in the current script for now (I used it in my old setup, but it was removed during some merge steps). It will be tuned later.

File: egs/chime5/s5/local/chain/tuning/run_tdnn_1e.sh [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+
+if [ $stage -le 14 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
+    utils/create_split_dir.pl \
+     /export/b0{3,4,5,6}/$USER/kaldi-data/egs/chime5-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
+  fi
+
+  steps/nnet3/chain/train.py --stage=$train_stage \
+    --cmd="$decode_cmd" \
```
Comment (kamo-naoyuki) [inline egs/chime5/s5/local/chain/tuning/run_tdnn_1e.sh]: Sorry, here. $decode_cmd should be replace to $cuda_cmd?

**Comment by jtrmal** [comment]
No, it shouldn't be cuda_cmd
Y.

**Comment by jtrmal** [comment]
Cuda_cmd was used for training Karels dnn, now the cuda during training is
used in different way.

**Comment by kamo-naoyuki** [comment]
ok, I understood.

File: egs/chime5/s5/local/train_lms_srilm.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (danpovey) [reply egs/chime5/s5/local/train_lms_srilm.sh]: @jtrmal, for these max-ent LMs, are they the ones that are actually chosen by the script?
I believe they are quite slow to estimate, and I'm not sure how they behave when pruned.
I'd prefer to get rid of the dependency if there isn't really a good reason to use them.

**Comment by jtrmal** [comment]
Yes, those are typically the ones used. The gain on the WER for this
dataset is around 0.2 % abs.
Also, I'm using it fairly commonly (for babel and others) and didn't run
into any problem.
But given the fact CHiME organizers' plan is to get is this merged as soon
as possible, I'm ok with deleting it.
y.

**Comment by danpovey** [comment]
it's OK, you can keep it if it helps.

**Comment by sw005320** [comment]
@jtrmal @danpovey I'm just reporting the current status. I removed stage 13 (lexicon update) to strictly follow the challenge regulation, and also added a location tag for future scoring. Now, I'm checking whether the recipe is working from scratch. Now it's working by the data cleaning stage, and will move to the chain model. I already confirmed it's working in the previous setup, and I think the check will be smoothly finished in the weekend. (I hope. I believe)

**Comment by sw005320** [comment]
@danpovey we've finished the recipe check, and updated the latest results. If this is no problem, please merge it.

**Comment by sw005320** [comment]
Thanks!!!!
